src/backend/distributed/shared_library_init.c

```diff
@@ -2090,6 +2090,21 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
+	DefineCustomBoolVariable(
+		"citus.skip_constraint_validation",
+		gettext_noop("Skip validation of constraints"),
+		gettext_noop("Validating constraints is a costly operation which effects Citus' "
+					 "performance negatively. With this GUC set to true, we skip "
+					 "validating them. Constraint validation can be redundant for some "
+					 "cases. For instance, when moving a shard, which has already "
+					 "validated constraints at the source; we don't need to validate "
+					 "the constraints again at the destination."),
+		&SkipConstraintValidation,
+		false,
+		PGC_USERSET,
```

Comment:
Without checking with one of our Postgres committers I don't think it's safe (security wise) to allow all users to set this option. Can we change its usage such that it only requries PGC_SUSET? e.g. by creating the foreign keys using the superuser.

Reply:
Foreign key creation can only be done by table owner.

>e.g. by creating the foreign keys using the superuser.

sounds risky (e.g. if the foreign key creation does an internal SELECT via SPI, can we get that SELECT to misbehave?)

Reply:
As discussed over a call, the security of the SELECT via SPI shouldn't be a problem in this case, since the whole purpose of this GUC is that it won't be executed.